clippy_lints/src/methods/filter_map_identity.rs

```diff
@@ -21,6 +21,14 @@ pub(super) fn check(cx: &LateContext<'_>, expr: &hir::Expr<'_>, filter_map_arg:
     if is_trait_method(cx, expr, sym::Iterator)
         && let Some(applicability) = is_identity(cx, filter_map_arg)
     {
+        if let Some((_, recv, _, _, _)) = method_call(expr)
+            && let Some((_, recv, _, _, _)) = method_call(recv)
+            && let hir::ExprKind::Array(arr) = recv.kind
```

Comment:
Use `let ExprKind::MethodCall(recv, ..) = expr.kind` rather than `method_call`.

Reply:
Fixed in https://github.com/rust-lang/rust-clippy/pull/13826/commits/d3d24e70ca59b2404510356e3be857045eab707b.